### PR TITLE
Error 404

### PR DESCRIPTION
**Pull request by von-moyo**
error 404
error 404(Poor maintenance)

built these two responsive error pages

### CODE REVIEW

File: src/features/error/error404NotFound/styles.module.scss [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    .errorButtons {
+      display: flex;
+      gap: 15px;
+
+      .errorBackBtn {
+        .backArrow {
+          width: 16px;
+        }
+
+        border: 1px solid #DDDFE3;
```
Comment (The-Anomaly) [inline src/features/error/error404NotFound/styles.module.scss:41]: Update this button styling and remove styles that the Button component have accounted for

File: src/features/error/error404PoorNetwork/index.tsx [hunk 2]
```diff
@@ -1,8 +1,35 @@
...
+    <div className={`appContainer ${styles.errorPage}`}>
+      <section className={styles.errorDetails}>
+        <h3 className={styles.errorName}>404 error</h3>
+        <h2 className={styles.errorTitle}>Poor Network</h2>
+        <h1 className={styles.errorDescription}>
+          Sorry, we can’t load this page right now due to poor network. Please
+          try again in a bit
+        </h1>
+        <div className={styles.errorButtons}>
+            <Button
```
Comment (The-Anomaly) [inline src/features/error/error404PoorNetwork/index.tsx:17]: The button type for this should be "outline" not transparent.

File: src/features/error/error404PoorNetwork/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    .errorButtons {
+      display: flex;
+      gap: 15px;
+
+      .errorBackBtn {
+        .backArrow {
+          width: 16px;
+        }
+
+        border: 1px solid #DDDFE3;
```
Comment (The-Anomaly) [inline src/features/error/error404PoorNetwork/styles.module.scss:41]: Update the styling for this button class and remove attributes that the Button component has accounted for

File: src/router/routeBuilder.ts
Comment (The-Anomaly) [inline src/router/routeBuilder.ts:1]: Undo this change, the path for 404 is a wildcard "*"

File: src/router/routes.ts [hunk 4]
```diff
@@ -13,6 +13,7 @@ export const Routes = {
   suppliers: "/suppliers",
   poorNetwork: "/poor-network",
   error500: "/error500",
+  error404: "/error404",
```
Comment (The-Anomaly) [inline src/router/routes.ts:16]: Remove this route.
A 404, Not found page shouldn't have a dedicated route.

Navigating to any unrecognized route (wrong url) matches the wildcard, "*", hence the page is displayed.

File: yarn.lock
Comment (The-Anomaly) [inline yarn.lock:1]: The changes to your package.json file affected the lock file.
I'm not expecting any changes here.

**Review by The-Anomaly — CHANGES_REQUESTED** [review]

File: src/assets/vectors/fileNotFound.png
[the remark this answers is not in the document]
Comment (von-moyo) [reply src/assets/vectors/fileNotFound.png:1]: oh yeah, that, i'll get rid of the png file.

File: src/features/error/error404NotFound/index.tsx [hunk 5]
```diff
@@ -1,8 +1,32 @@
...
+  return (
+    <div className={`appContainer ${styles.errorPage}`}>
+      <section className={styles.errorDetails}>
+        <h3 className={styles.errorName}>404 error</h3>
+        <h2 className={styles.errorTitle}>We can't find that page</h2>
+        <h1 className={styles.errorDescription}>
+          Sorry, the page you're looking for doesn't exist or has beeen moved.
+        </h1>
+        <div className={styles.errorButtons}>
+          <Button type="transparent" className={styles.errorBackBtn}>
```
[the remark this answers is not in the document]
Comment (von-moyo) [reply src/features/error/error404NotFound/index.tsx:16]: okay, will fix it.

File: package.json
[the remark this answers is not in the document]
Comment (von-moyo) [reply package.json:1]: what changes exactly?

File: src/router/routes.ts (hunk 4 above)
Comment (von-moyo) [reply src/router/routes.ts:16]: oh cool cool.

**Review by The-Anomaly — DISMISSED** [review]

**Pull request merged**